src/lib/ndpi_utils.c

```diff
@@ -1439,7 +1439,9 @@ int ndpi_flow2json(struct ndpi_detection_module_struct *ndpi_struct,
     break;
   }
 
-  return(ndpi_dpi2json(ndpi_struct, flow, l7_protocol, serializer));
+  int res = ndpi_dpi2json(ndpi_struct, flow, l7_protocol, serializer);
+  ndpi_term_serializer(serializer);
+  return res;
```

Comment:
I don't use/know the serialization code, so this may be a silly question: why `serializer` is (still) a function parameter if it is initialized/de-initialized in the function itself?
Am I missing something obvious?

Reply:
Oh yes, you're right. There's no need to pass `serializer` as a parameter anymore.